packages/core/src/components/icon/icon.tsx

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ * Licensed under the BSD-3 License as modified (the “License”); you may obtain a copy
+ * of the license at https://github.com/palantir/blueprint/blob/master/LICENSE
+ * and https://github.com/palantir/blueprint/blob/master/PATENTS
+ */
+
+import * as classNames from "classnames";
+import * as React from "react";
+
+import { Classes, IIntentProps, IProps } from "../../common";
+import { ICON_DEPRECATED_NAMES } from "../../common/errors";
+import { isNodeEnv } from "../../common/utils";
+import { IconName } from "../../generated/iconName";
+
+export { IconName }
+
+export interface IIconProps extends IIntentProps, IProps {
+    /**
+     * Name of the icon (the part after `pt-icon-`).
+     * If `undefined`, this component will render nothing.
+     *
+     * The `pt-icon-` prefix is currently supported in icon names but will be removed in v2.0.
+     */
+    iconName: IconName | undefined;
+
+    /**
+     * Size of the icon.
+     * Blueprint provides each icon in two sizes: 16px and 20px. The keyword `"inherit"` will
+     * render a 20px icon but inherit `font-size` from its parent.
+     * @default 16
+     */
+    iconSize?: 16 | 20 | "inherit";
```

Comment:
not a fan of requiring users write out `iconSize={undefined}`, cuz it defaults to `16`.
i thought about using a different keyword, but the string just made the most sense to me cuz it sets appropriate...

### CSSpectations 👊